### DISCUSSION TITLE
What effect does the overflow have?

### DISCUSSION THREAD

**Discussion by kaphula** [DioxusLabs/taffy#647] [Q&A] [answered]
I don't understand what is the concrete effect of specifying `overflow` in `Style`. Can someone explain this to me? My thought was that setting `overflow` to `Scroll` would automatically adjust the node's scrollbar to specified `scrollbar_size` from `0` when the node's children content size surpasses the node's size, but that does not seem to be the case. If I set `scrollbar_size` style to a node, it is always active, eating a portion from the node's space. So what is the point of having `overflow` style setting then?

**Comment by nicoburns** [comment, marked as answer]
We will hopefully implement `overflow: auto` (which behaves as you expected `scroll` to in the future).

The biggest thing at the moment is that `overflow: hidden` and `overflow: scroll` disable the "automatic minimum size" for Flexbox and CSS Grid items. Also, `overflow: scroll` enables the `scrollbar_size` style. That's a little redundant (you could just set it to 0), but means that our API is 1:1 with the web API (Taffy's `overflow` property behaves like the web overflow property). Finally, it affects the computed `content_size` (if overflow is `visible` then overflowing children are included in that nodes size for scroll width/height calculation).
